pkg/controller/association/controller/apm_kibana.go

```diff
@@ -69,7 +81,58 @@ func getKibanaExternalURL(c k8s.Client, assoc commonv1.Association) (string, err
 		serviceName = kbv1.HTTPService(kb.Name)
 	}
 	nsn := types.NamespacedName{Namespace: kb.Namespace, Name: serviceName}
-	return association.ServiceURL(c, nsn, kb.Spec.HTTP.Protocol())
+
+	// Get Kibana base path if configured
+	basePath, err := getKibanaBasePath(c, kb)
+	if err != nil {
+		return "", err
+	}
+	return association.ServiceURL(c, nsn, kb.Spec.HTTP.Protocol(), basePath)
+}
+
+func getKibanaBasePath(client k8s.Client, kb kbv1.Kibana) (string, error) {
+	// base path set as ENV variable takes precedence over the one set in the Kibana config secret
+	kbBasePath, err := getKibanaBasePathFromEnv(client, kb)
+	if err != nil {
+		return "", fmt.Errorf("failed to get Kibana base path from environment: %w", err)
+	}
+
+	if kbBasePath != "" {
+		return kbBasePath, nil
+	}
+
+	if kb.Spec.Config == nil {
+		return "", nil
+	}
+
+	// Get Kibana config secret to extract the basepath.
+	// We are not using the Kibana CRD here for the basepath to optimize for the case where the desired and current state may differ, so we're choosing the current state to minimize any transient errors.
```

Comment:
I my testing I still see a fair amount of reconciliation errors with this approach. I believe this is due to the configuration being generated almost at the same time as the extraction of the association configuration, while you still might have old Kibana pods running which won't re-read their configuration. 

This makes me think that we should just go with what is in the Kibana custom resource. This would also address another issue I have with the current approach which is that we duplicate the implementation between the association configuration and the readiness probe. I would suggest to just use one implemenation shared between the two use cases (I think it should be possible to import the one from the Kibana controller here, which is based on the CR as the source of truth)

Reply:
I changed it to use a single method for both